oauth2_provider/oauth2_backends.py

```diff
@@ -136,6 +156,24 @@ def verify_request(self, request, scopes):
         return valid, r
 
 
+class JSONOAuthLibCore(OAuthLibCore):
+    """
+    Extends the default OAuthLibCore to parse correctly requests with application/json Content-Type
+    """
+    def _extract_body(self, request):
+        """
+        Extracts the JSON body from the Django request object
+        :param request: The current django.http.HttpRequest object
+        :return: provided POST parameters "urlencodable"
+        """
+        try:
+            body = json.loads(request.body.decode('utf-8')).items()
+        except ValueError:
+            body = ""
```

Comment:
The issue #127 is caused because the extracted body was empty for a wrong request.POST decoding in ``application/json`` requests. In this case I maintain the previous behavior while we can raise an exception if the request is not correctly formatted as a JSON. Removing this try-except, leads to an internal server error instead of a bad request.